library/core/src/iter/traits/accum.rs

```diff
@@ -27,6 +31,10 @@ pub trait Sum<A = Self>: Sized {
 /// [`product()`]: Iterator::product
 /// [`FromIterator`]: iter::FromIterator
 #[stable(feature = "iter_arith_traits", since = "1.12.0")]
+#[rustc_on_unimplemented(
+    message = "a value of type `{Self}` cannot be made by for the product of an iterator with elements of type `{A}`",
+    label = "value of type `{Self}` cannot be made with the product a `std::iter::Iterator<Item={A}>`"
```

Comment:
This isn't grammatical I think, though I don't know if there's an analogous verb to "sum" for "multiplying all together"... Maybe "multiplying" works though.

Reply:
I think that is a reasonable wording here.

Reply:
@aDotInTheVoid I can r+ once you've changed the wording here to what you drafted in the comment :)

Reply:
Updated